pkg/volume/glusterfs/glusterfs.go

```diff
@@ -0,0 +1,233 @@
+/*
+Copyright 2015 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package glusterfs
+
+import (
+	"math/rand"
+	"os"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/types"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util/exec"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util/mount"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/volume"
+	"github.com/golang/glog"
+)
+
+// This is the primary entrypoint for volume plugins.
+func ProbeVolumePlugins() []volume.VolumePlugin {
+	return []volume.VolumePlugin{&glusterfsPlugin{nil}}
+}
+
+type glusterfsPlugin struct {
+	host volume.VolumeHost
+}
+
+var _ volume.VolumePlugin = &glusterfsPlugin{}
+
+const (
+	glusterfsPluginName = "kubernetes.io/glusterfs"
+)
+
+func (plugin *glusterfsPlugin) Init(host volume.VolumeHost) {
+	plugin.host = host
+}
+
+func (plugin *glusterfsPlugin) Name() string {
+	return glusterfsPluginName
+}
+
+func (plugin *glusterfsPlugin) CanSupport(spec *api.Volume) bool {
+	if spec.VolumeSource.Glusterfs != nil {
+		return true
+	}
+	return false
+}
+
+func (plugin *glusterfsPlugin) GetAccessModes() []api.AccessModeType {
```

Comment:
loop in @markturansky 
there is a followup on the persistent volume front - once gluster is made to be able to handle it.